engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1011,7 +1012,7 @@ private void execute() throws YarnException, IOException
             numCompletedContainers.incrementAndGet();
             LOG.info("Container completed successfully." + ", containerId=" + containerStatus.getContainerId());
             // Reset counter for node failure, if exists
-            String hostname = allocatedContainer.container.getNodeId().getHost();
+            String hostname = allocatedContainer.host;
```

Comment:
Yes, it looks like it can be `null` when exitStatus is 0. It will be good to understand how a container can exit with the exit status 0 without being added to the `allocatedContainers` or was it already removed from the list.